### PR TITLE
K8SPSMDB-390: Fix RS initialization

### PR DESCRIPTION
**Pull request by blez**
[K8SPSMDB-390](https://jira.percona.com/browse/K8SPSMDB-390) [&#10088;?&#10089;](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

### CODE REVIEW

File: pkg/controller/perconaservermongodb/mgo.go [hunk 1]
```diff
@@ -51,6 +51,10 @@ func (r *ReconcilePerconaServerMongoDB) reconcileCluster(cr *api.PerconaServerMo
 		}
 
 		return clusterError, errors.Wrap(err, "dial:")
+	} else if !cr.Status.Replsets[replset.Name].Initialized {
+		// this can happended when cluster was initialized but status update failed
+		cr.Status.Replsets[replset.Name].Initialized = true
```
Comment (delgod) [inline pkg/controller/perconaservermongodb/mgo.go:56]: should we also add lines line 43-48 here?

**Review by delgod — CHANGES_REQUESTED** [review]

File: pkg/controller/perconaservermongodb/mgo.go (hunk 1 above)
Comment (blez) [reply pkg/controller/perconaservermongodb/mgo.go:57]: https://github.com/percona/percona-server-mongodb-operator/pull/558/commits/2d96cf9aa5411f6a791dd9a688a561f0f4ddd372

**Review by delgod — APPROVED** [review]

**Comment by JNKPercona** [comment]
PSMDB operator docker - perconalab/percona-server-mongodb-operator:PR-558-d94ac967

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| one-pod | passed |
| init-deploy | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| demand-backup | passed |
| limits | passed |
| scaling | passed |
| scheduled-backup | passed |
| monitoring-2-0 | passed |
| security-context | passed |
| operator-self-healing | passed |
| demand-backup-sharded | passed |
| rs-shard-migration | passed |
| arbiter | passed |
| upgrade | passed |
| service-per-pod | passed |
| upgrade-sharded | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |
| data-sharded | passed |

**Comment by JNKPercona** [comment]
PSMDB operator docker - perconalab/percona-server-mongodb-operator:PR-558-2d96cf9a

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| one-pod | passed |
| storage | passed |
| upgrade-consistency | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| self-healing | passed |
| limits | passed |
| scaling | passed |
| arbiter | passed |
| scheduled-backup | passed |
| security-context | passed |
| operator-self-healing | passed |
| service-per-pod | passed |
| demand-backup-sharded | passed |
| rs-shard-migration | passed |
| liveness | passed |
| upgrade | passed |
| smart-update | passed |
| upgrade-sharded | passed |
| version-service | passed |
| users | passed |
| data-sharded | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| storage | passed |
| init-deploy | passed |
| self-healing | passed |
| monitoring-2-0 | failed |
| limits | passed |
| scaling | passed |
| operator-self-healing | passed |
| security-context | passed |
| rs-shard-migration | passed |
| arbiter | passed |
| service-per-pod | failed |
| liveness | passed |
| smart-update | failed |
| version-service | failed |

**Review by tplavcic — APPROVED** [review]

**Review by qJkee — APPROVED** [review]

**Comment by JNKPercona** [comment]
PSMDB operator docker - perconalab/percona-server-mongodb-operator:PR-558-cfa7fad5

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| monitoring-2-0 | passed |
| storage | passed |
| self-healing | passed |
| operator-self-healing | passed |
| upgrade-consistency | passed |
| init-deploy | passed |
| limits | passed |
| scaling | passed |
| security-context | passed |
| rs-shard-migration | passed |
| demand-backup | passed |
| arbiter | passed |
| service-per-pod | passed |
| scheduled-backup | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| demand-backup-sharded | passed |
| upgrade | passed |
| users | passed |
| upgrade-sharded | passed |
| data-sharded | passed |

**Pull request merged**